### PR TITLE
Do not reuse adminClient instance

### PR DESCRIPTION
**Pull request by am-wr**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@am-wr Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=804)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=804) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@am-wr Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=804)!

File: spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderHealthIndicator.java [hunk 1]
```diff
@@ -77,57 +73,38 @@
...
-						builder.status(Status.UNKNOWN);
-					}
-					else {
-						boolean up = true;
-						for (KafkaStreams kStream : kafkaStreamsRegistry.getKafkaStreams()) {
-							up &= kStream.state().isRunning();
-							builder.withDetails(buildDetails(kStream));
-						}
-						builder.status(up ? Status.UP : Status.DOWN);
+		try (AdminClient adminClient = AdminClient.create(this.adminClientProperties)) {
```
Comment (sobychacko) [inline spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderHealthIndicator.java]: This still has the problem of creating `AdminClient` on each request. We will fix it when we merge the PR.

**Comment by sobychacko** [comment]
Thanks for the PR! This is merged upstream with an extra commit on top of it to reuse the `adminClient`.  More feedback and PR's welcomed!

File: spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderHealthIndicator.java (hunk 1 above)
Comment (am-wr) [reply spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderHealthIndicator.java]: That is true. FYO: I have noticed this `synchronized (this.adminClient)`. Synchronizing on non-final is not recommended.
I didn't know if `AdminClient` is capable of reconnection. If it does not - there should be a way to determine if needs to be recreated/reconnected.  
This object looks very expensive as starts a new tread every time  (that can be used as a potential attack vector).  Another point to consider - caching status for some time (even for 100ms).

**Pull request closed without merging**